jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt

```diff
@@ -0,0 +1,305 @@
+                            debuggerLifetimeDefinition.terminate()
+                        }
+                    })
+
+                    workerModel.targetExited.advise(debuggerLifetime) {
+                        logger.trace { "Target exited" }
+                        // We should try to kill deployment there because it's already stopped,
+                        // just notify debugger session about termination via its process handler.
+                        processHandler.notifyProcessDestroyed(it.exitCode ?: 0)
+                    }
+
+                    promise.setResult(
+                        createAndStartSession(
+                            executionConsole = console,
+                            env = environment,
+                            sessionLifetime = debuggerLifetime,
+                            processHandler = processHandler,
+                            protocol = protocol,
+                            sessionModel = sessionModel,
+                            outputEventsListener = object : IDebuggerOutputListener {})
+                    )
+
+                    return@initialized true
+                }
+            } catch (t: Throwable) {
+                debuggerLifetimeDefinition.terminate()
+                promise.setError(t)
+            }
+            return@connected true
+        }
+
+        val checkDebuggerTask = Timer("Debugger Worker launch timer", true).schedule(debuggerAttachTimeoutMs) {
+            if (debuggerLifetimeDefinition.isAlive && !protocol.wire.connected.value) {
+                debuggerLifetimeDefinition.terminate()
+                promise.setError(message("lambda.debug.process.start.timeout"))
+            }
+        }
+
+        debuggerLifetime.onTermination {
+            checkDebuggerTask.cancel()
+        }
+
+        return promise
+    }
+
+    private fun createNetCoreStartInfo(state: SamRunningState): DotNetCoreExeStartInfo =
+        DotNetCoreExeStartInfo(
+            dotNetCoreInfo = DotNetCoreInfo(dotnetCorCliPath, null),
+            exePath = mockBootstrapsPath,
+            workingDirectory = "",
+            arguments = state.settings.handler,
+            environmentVariables = emptyList(),
+            runtimeArguments = null,
+            executeAsIs = false,
+            useExternalConsole = false,
+            needToBeInitializedImmediately = true
+        )
+
+    private fun createAndStartSession(
+        executionConsole: ExecutionConsole,
+        env: ExecutionEnvironment,
+        sessionLifetime: Lifetime,
+        processHandler: ProcessHandler,
+        protocol: IProtocol,
+        sessionModel: DotNetDebuggerSessionModel,
+        outputEventsListener: IDebuggerOutputListener
+    ): XDebugProcessStarter {
+        XDebuggerManager.getInstance(env.project)
+
+        val consoleKind = ConsoleKind.ExternalConsole
+
+        if (consoleKind != ConsoleKind.Normal) {
```

Comment:
When would this if statement change? `(ConsoleKind.ExternalConsole != ConsoleKind.Normal`)

Reply:
Fixed. Removed useless check for consoleKind